### PR TITLE
HHH-11147 : fixes batch fetching bytecode-enhanced proxies

### PR DESCRIPTION
**Pull request by dreab8**
This is the Gail's https://github.com/hibernate/hibernate-orm/pull/2930 without the failing tests related with the initialisation of an non existing entity.

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/event/internal/DefaultLoadEventListener.java [hunk 1]
```diff
@@ -299,6 +299,21 @@ private Object proxyOrLoad(
 					event.getSession()
 			);
 
+			// IMPLEMENTATION NOTE:
+			// The entity's EntityKey may already have been added to the BatchFetchQueue
+			// by ManyToOneType#scheduleBatchLoadIfNeeded.
+			// In the case that the entity is loaded by Session#load or Session#getReference,
+			// it will not be in the BatchFetchQueue, so go ahead and add it.
+			persistenceContext.getBatchFetchQueue().addBatchLoadableEntityKey( keyToLoad );
```
Comment (Sanne) [inline hibernate-core/src/main/java/org/hibernate/event/internal/DefaultLoadEventListener.java]: I suppose this section should also check for `org.hibernate.persister.entity.EntityPersister#isBatchLoadable` ?

Comment (sebersole) [reply hibernate-core/src/main/java/org/hibernate/event/internal/DefaultLoadEventListener.java]: Why?  Its done in `#addBatchLoadableEntityKey` - no need to do it twice

Comment (Sanne) [reply hibernate-core/src/main/java/org/hibernate/event/internal/DefaultLoadEventListener.java]: oh right.

I got confused, but I was checking something odd: it seems like this patch is able to reduce the number of SQL statements being executed even when batching is not enabled.

Trying to figure out why.. ideas?

**Comment by Sanne** [comment]
merged #2933 instead

**Pull request closed without merging**